praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/service/PraktikumsstellenService.java

```diff
@@ -149,6 +151,23 @@ public void deleteAusbildungsPraktikumsstelle(UUID praktikumsstellenId) {
         }
     }
 
+    public void updateAusbildungsPraktikumsstelle(UUID praktikumsstellenId,
+            CreateAusbildungsPraktikumsstelleWithMeldezeitraumAndAssignedNWKDto praktikumsstelleDto) {
+        if (ausbildungsPraktikumsstellenRepository.existsById(praktikumsstellenId)) {
+            ausbildungsPraktikumsstellenRepository.save(praktikumsstellenMapper.toEntity(praktikumsstellenId, praktikumsstelleDto));
+        } else {
+            throw new ResourceNotFoundException("Praktikumsstelle nicht gefunden.");
+        }
+    }
+
+    public void updateStudiumsPraktikumsstelle(UUID praktikumsstellenId, CreateStudiumsPraktikumsstelleWithMeldezeitraumAndAssignedNwkDto praktikumsstelleDto) {
+        if (studiumsPraktikumsstellenRepository.existsById(praktikumsstellenId)) {
+            studiumsPraktikumsstellenRepository.save(praktikumsstellenMapper.toEntity(praktikumsstellenId, praktikumsstelleDto));
+        } else {
+            throw new ResourceNotFoundException("Praktikumsstelle nicht gefunden.");
+        }
+    }
```

Comment:
The rules stated in the issue should not only be enforced in the frontend but also in the backend. With the current code it would be possible to change the ausbildungsrichtung or other data that should not be changed when a NWK has already been assigned